interview-questions.md

```diff
@@ -7,60 +7,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is an instance variable in Ruby? How is it different from other variables?
 
-  Your answer:
+  Your answer: An instance variable is when youre creating an instance inside the varaible. Calling on the instances or instantiating is another term when you are calling upon an action. 
 
-  Researched answer:
+  Researched answer: In object-oriented programming with classes, an instance variable is a variable defined in a class, for which each instantiated object of the class has a separate copy, or instance.
 
 
 
 2. What is a block in Ruby?
 
-  Your answer:
+  Your answer: I remember the blocks challenge while i was driving and not navigating that you needed a d0 in the process. 
 
-  Researched answer:
+  Researched answer:Ruby has anonymous functions similar to JavaScript. They are called blocks. Blocks can be created with the do/end keywords OR with {}. Blocks are code that is passed to a method.
 
 
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: From the note-taking I did under this topic I remember writing over and over that its when the return is automated once you return something in ruby without typing return.
+
+  Researched answer: Ruby has an implicit return, which means the value of the last line of a method is automatically returned without using the keyword return.
+
 
-  Researched answer:
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: From week-3-assessment this was apart of the looking ahead vocabulary. I remember the acronym OOP which was creating objects that contain data.
 
-  Researched answer:
+  Researched answer: Ruby has anonymous functions similar to JavaScript. They are called blocks. Blocks can be created with the do/end keywords OR with {}. Blocks are code that is passed to a method. The difference from OOP to functional programming 
 
 
 
 5. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: The difference is the object is called as an instance of a class. So say the bike is the class, the wheels, speed, model, are objects
 
-  Researched answer:
+  Researched answer: An object is a unit of data. A class is what kind of data it is.
 
 
 
 6. STRETCH: What is `attr_accessor`?
 
-  Your answer:
+  Your answer: This allows the child classs to extract the information from the parent class. 
```

Comment:
that would be super() -- although physically close in our code but they do different things.